cartography/intel/aws/ecr.py

```diff
@@ -63,44 +63,47 @@ def load_ecr_repositories(neo4j_session, data, region, current_aws_account_id, a
 @timeit
 def load_ecr_repository_images(neo4j_session, repo_data, region, aws_update_tag):
     query = """
-    MERGE (repo_image:ECRRepositoryImage{id: {RepositoryImageUri}})
-    ON CREATE SET repo_image.firstseen = timestamp()
-    SET repo_image.lastupdated = {aws_update_tag}, repo_image.tag = {ImageTag},
-        repo_image.uri = {RepositoryImageUri}
-    WITH repo_image
-
-    MERGE (image:ECRImage{id: {ImageDigest}})
-    ON CREATE SET image.firstseen = timestamp(), image.digest = {ImageDigest}
-    SET image.lastupdated = {aws_update_tag},
-    image.region = {Region}
-    WITH repo_image, image
-    MERGE (repo_image)-[r1:IMAGE]->(image)
-    ON CREATE SET r1.firstseen = timestamp()
-    SET r1.lastupdated = {aws_update_tag}
-    WITH repo_image
-
-    MATCH (repo:ECRRepository{uri: {RepositoryUri}})
-    MERGE (repo)-[r2:REPO_IMAGE]->(repo_image)
-    ON CREATE SET r2.firstseen = timestamp()
-    SET r2.lastupdated = {aws_update_tag}
+    UNWIND {RepoList} as repo_item
+        UNWIND repo_item.repo_images as repo_img
+            MERGE (ri:ECRRepositoryImage{id: repo_item.repo_uri + ":" + repo_img.imageTag})
+            ON CREATE SET ri.firstseen = timestamp()
+            SET ri.lastupdated = {aws_update_tag},
+            ri.tag = repo_img.imageTag,
+            ri.uri = repo_item.repo_uri + ":" + repo_img.imageTag
+            WITH ri, repo_img, repo_item
+
+            MERGE (img:ECRImage{id: repo_img.imageDigest})
+            ON CREATE SET img.firstseen = timestamp(),
+            img.digest = repo_img.imageDigest
+            SET img.lastupdated = {aws_update_tag},
+            img.region = {Region}
```

Comment:
Ditto indentation

Is an image in a single region or can it be in multiple regions? If it's the latter then we should make this a list instead of overwriting the value.

Reply:
Single region, since an image lives in a repository that is in a single region.

Come to think of it, we might not actually need a region field for the image since the repository has it defined already but ./shrug.